modo/cram.py

```diff
@@ -10,20 +10,66 @@
 from rdflib import Graph
 import modo_schema.datamodel as model
 
+import os, sys, re
+import htsget
 from .helpers import parse_region
 
 
 def slice_cram(path: str, region: str) -> Iterator[AlignedSegment]:
     """Return an iterable slice of the CRAM file."""
 
-    chrom, start, stop = parse_region(region)
+    chrom, start, end = parse_region(region)
+    if start == "":
+        start = None
+    else:
+        start = int(start)
+    if end == "":
+        end = None
+    else:
+        end = int(end)
     cramfile = AlignmentFile(path, "rc")
 
-    iter = cramfile.fetch(chrom, start, stop)
+    iter = cramfile.fetch(chrom, start, end)
 
     return iter
 
 
+def slice_remote_cram(
+    url: str, region: str = None, output_filename: str = None
+):
+    """Stream or write to a local file a slice of a remote CRAM file"""
+
+    url = str(
+        Path(url).with_suffix("")
+    )  # htsget.get needs URL without the file extension
+    reference_name, start, end = parse_region(region)
+    if start == "":
+        start = None
+    else:
+        start = int(start)
+    if end == "":
+        end = None
+    else:
+        end = int(end)
+
+    if output_filename:
+        with open(output_filename, "wb") as output:
+            htsget.get(
+                url, output, reference_name, start, end, data_format="cram"
```

Comment:
**Suggestion(minor)**: With this number of arguments I think it is safer to use named arguments even if they read repetitive:
 ```suggestion
            htsget.get(
                url = url, 
                output = output,
                reference_name = reference_name, 
                start = start, 
                end = end, 
                data_format="cram"
```